projects/packages/waf/src/class-waf-runtime.php

```diff
@@ -435,95 +457,93 @@ public function unset_var( $key ) {
 	}
 
 	/**
-	 * Meta.
+	 * A cache of metadata about the incoming request.
 	 *
-	 * @param string $key Key.
-	 * @param string $prop Prop.
+	 * @param string $key The type of metadata to request ('headers', 'remote_addr', etc.).
 	 */
-	public function meta( $key, $prop = false ) {
+	public function meta( $key ) {
 		if ( ! isset( $this->metadata[ $key ] ) ) {
 			$value = null;
 			switch ( $key ) {
 				case 'headers':
-					$value = array();
```

Comment:
Moved most of the logic for getting data out of PHP superglobals (`$_SERVER`, `$_GET`, etc.) into the `Waf_Request` so that it's easier to mock and test.